components/helm-broker/internal/controller/addonsconfiguration_controller.go

```diff
@@ -0,0 +1,500 @@
+	for _, b := range existingBundles {
+		if r.hasBundleDeleted(b, addonsList) {
+			r.log.Infof("- delete bundle %s from storage", b.Name)
+			err := r.bundleStorage.Remove(internal.ClusterWide, internal.BundleName(b.Name), b.Version)
+			if err != nil {
+				return exerr.Wrapf(err, "while removing bundle %s/%q", b.Name, b.Version)
+			}
+			err = r.chartStorage.Remove(internal.ClusterWide, internal.ChartName(b.Name), b.Version)
+			if err != nil {
+				return exerr.Wrapf(err, "while removing chart %s/%q", b.Name, b.Version)
+			}
+		}
+	}
+	return nil
+}
+
+func (r *ReconcileAddonsConfiguration) hasBundleDeleted(b *internal.Bundle, list *addonsv1alpha1.AddonsConfigurationList) bool {
+	for _, cfgs := range list.Items {
+		for _, repo := range cfgs.Status.Repositories {
+			for _, add := range repo.Addons {
+				if string(b.Name) == add.Name && b.Version.Original() == add.Version {
+					return false
+				}
+			}
+		}
+	}
+	return true
+}
+
+func (r *ReconcileAddonsConfiguration) ensureBroker(addon *addonsv1alpha1.AddonsConfiguration) error {
+	exist, err := r.brokerFacade.Exist(addon.Namespace)
+	if err != nil {
+		return exerr.Wrapf(err, "while checking if ServiceBroker exist in namespace %s", addon.Namespace)
+	}
+	if !exist {
+		// status
+		if err := r.brokerFacade.Create(addon.Namespace); err != nil {
+			return exerr.Wrapf(err, "while creating ServiceBroker for AddonConfiguration %s/%s", addon.Name, addon.Namespace)
+		}
+	} else if r.syncBroker {
+		if err := r.brokerSyncer.SyncServiceBroker(addon.Namespace); err != nil {
+			return exerr.Wrapf(err, "while syncing ServiceBroker for AddonConfiguration %s/%s", addon.Name, addon.Namespace)
+		}
+	}
+	return nil
+}
+
+func (r *ReconcileAddonsConfiguration) createAddons(URL string) ([]*addons.AddonController, error) {
+	adds := []*addons.AddonController{}
+
+	// fetch repository index
+	index, err := r.bundleProvider.GetIndex(URL)
+	if err != nil {
+		return adds, exerr.Wrap(err, "while reading repository index")
+	}
+
+	// for each repository entry create addon
+	for _, entries := range index.Entries {
+		for _, entry := range entries {
+			addon := addons.NewAddon(string(entry.Name), string(entry.Version), URL)
+
+			completeBundle, err := r.bundleProvider.LoadCompleteBundle(entry)
+			if bundle.IsFetchingError(err) {
+				addon.FetchingError(err)
+				adds = append(adds, addon)
+				logrus.Errorf("while fetching addon: %s", err)
+				continue
+			}
+			if bundle.IsLoadingError(err) {
+				addon.LoadingError(err)
+				adds = append(adds, addon)
+				logrus.Errorf("while loading addon: %s", err)
+				continue
+			}
+
+			addon.ID = string(completeBundle.Bundle.ID)
+			addon.Bundle = completeBundle.Bundle
+			addon.Charts = completeBundle.Charts
+
+			adds = append(adds, addon)
+		}
+	}
+
+	return adds, nil
+}
+
+func (r *ReconcileAddonsConfiguration) saveBundle(namespace internal.Namespace, repositories *addons.RepositoryCollection) error {
+	for _, addon := range repositories.ReadyAddons() {
```

Comment:
Before saving bundles/charts check if `AddonsConfiguration` is in a ready state like we discussed.

Reply:
done